platform/ios/MapLibre.docc/WebAPIDataExampleUIKit.md

```diff
@@ -0,0 +1,224 @@
+    }
+
+    // MARK: - Data fetching and parsing
+
+    func fetchPoints(withCompletion completion: @escaping (([MLNPointFeature]) -> Void)) {
+        // Wikidata query for all lighthouses in the United States: http://tinyurl.com/zrl2jc4
+        let query = "SELECT DISTINCT ?item " +
+            "?itemLabel ?coor ?image " +
+            "WHERE " +
+            "{ " +
+            "?item wdt:P31 wd:Q39715 . " +
+            "?item wdt:P17 wd:Q30 . " +
+            "?item wdt:P625 ?coor . " +
+            "OPTIONAL { ?item wdt:P18 ?image } . " +
+            "SERVICE wikibase:label { bd:serviceParam wikibase:language \"en\" } " +
+            "} " +
+            "ORDER BY ?itemLabel"
+
+        let characterSet = NSMutableCharacterSet()
+        characterSet.formUnion(with: CharacterSet.urlQueryAllowed)
+        characterSet.removeCharacters(in: "?")
+        characterSet.removeCharacters(in: "&")
+        characterSet.removeCharacters(in: ":")
+
+        let encodedQuery = query.addingPercentEncoding(withAllowedCharacters: characterSet as CharacterSet)!
+
+        let request = URLRequest(url: URL(string: "https://query.wikidata.org/sparql?query=\(encodedQuery)&format=json")!)
+
+        URLSession.shared.dataTask(with: request, completionHandler: { data, _, error in
+            guard error == nil else {
+                preconditionFailure("Failed to load GeoJSON data: \(error!)")
+            }
+
+            guard
+                let data,
+                let json = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: AnyObject],
+                let results = json["results"] as? [String: AnyObject],
+                let items = results["bindings"] as? [[String: AnyObject]]
+            else {
+                preconditionFailure("Failed to parse GeoJSON data")
+            }
+
+            DispatchQueue.main.async {
+                completion(self.parseJSONItems(items: items))
+            }
+        }).resume()
+    }
+
+    func parseJSONItems(items: [[String: AnyObject]]) -> [MLNPointFeature] {
+        var features = [MLNPointFeature]()
+        for item in items {
+            guard let label = item["itemLabel"] as? [String: AnyObject],
+                  let title = label["value"] as? String else { continue }
+            guard let coor = item["coor"] as? [String: AnyObject],
+                  let point = coor["value"] as? String else { continue }
+
+            let parsedPoint = point.replacingOccurrences(of: "Point(", with: "").replacingOccurrences(of: ")", with: "")
+            let pointComponents = parsedPoint.components(separatedBy: " ")
+            let coordinate = CLLocationCoordinate2D(latitude: Double(pointComponents[1])!, longitude: Double(pointComponents[0])!)
+            let feature = MLNPointFeature()
+            feature.coordinate = coordinate
+            feature.title = title
+            // A feature’s attributes can used by runtime styling for things like text labels.
```

Comment:
I would consider making the examples a bit shorter, 200 line example is a bit long to read.

Reply:
I can do another pass and chop them up later.